x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityLicensingAndFeatureUsageRestIT.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.remotecluster;
+
+import org.apache.http.client.methods.HttpDelete;
+import org.apache.http.client.methods.HttpGet;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.ResponseException;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.core.Strings;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.test.cluster.ElasticsearchCluster;
+import org.elasticsearch.test.rest.ObjectPath;
+import org.junit.ClassRule;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestRule;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.in;
+import static org.hamcrest.Matchers.is;
+
+public class RemoteClusterSecurityLicensingAndFeatureUsageRestIT extends AbstractRemoteClusterSecurityTestCase {
+
+    private static final AtomicReference<Map<String, Object>> API_KEY_MAP_REF = new AtomicReference<>();
+    public static final String SECURITY_CROSS_CLUSTER_ACCESS_FEATURE_NAME = "security-cross-cluster-access";
+
+    static {
+        fulfillingCluster = ElasticsearchCluster.local()
+            .name("fulfilling-cluster")
+            .nodes(1)
+            .apply(commonClusterConfig)
```

Comment:
`commonClusterConfig` configures trial license for the test cluster so it is eligible for the RCS feature when the remote cluster gets configured. I think we should instead starting with a lower license, e.g. basic, to show that the feature goes through the flow of "not allowed" -> "allowed" (start trial) -> "not allowed again" (delete license). 

Note there is a broader issue in error reporting for remote connection failures. When establishing the connection, underlying exceptions are swallowed and we get only `NoSeedNodeLeftException` in both logs and API response. This is the case for license issue as well, i.e. if the license is incompatible before configuring the remote cluster, we won't be able to get the nicer `license non-compliant` exception, but instead a "no_seed_node_left_exception". This is definitely not ideal and we should look into improving it. But improving the exact error reporting can be a separate issue from proving the feature does _not_ work without proper license since the error reporting affects more than just license, e.g. it also impacts when the cluster credentials is not configured.

Reply:
I've refactored the test now to test the flow: `"not allowed" -> "allowed" (start trial) -> "not allowed again"` . Thanks for the suggestion. 

As you already pointed out, I get the `NoSeedNodeLeftException` when running a CCS with unsupported license and the actual cause gets ignored. I left a `TODO` in the test which should be addressed once we improve the error reporting.